lib/app.js

```diff
@@ -202,3 +207,33 @@ function getProperty(target, prop) {
   }
   return member;
 }
+
+
+function bindMessenger(Application, agent) {
+  const agentMessenger = agent.messenger;
+  return class MeesengerApplication extends Application {
+    constructor(options) {
+      super(options);
+
+      const appMessenger = this.messenger;
+      agentMessenger.send = new Proxy(agentMessenger.send, {
+        apply(target, thisArg, [ action, data ]) {
+          setImmediate(() => appMessenger._onMessage({ action, data }));
+        },
+      });
+    }
+    get messenger() {
+      return this[MESSENGER];
+    }
+    set messenger(m) {
+      m.send = new Proxy(m.send, {
+        apply(target, thisArg, [ action, data ]) {
+          setImmediate(() => agentMessenger._onMessage({ action, data }));
+        },
+      });
+      this[MESSENGER] = m;
+    }
+
+    get [Symbol.for('egg#eggPath')]() { return path.dirname(__dirname); }
```

Comment:
For a better understanding, this is a requirement of  egg-core module defining how a framework should be implemented. The purpose of this design is for a way of resolving plugins paths.

So this is for standards consideration, but in fact, it takes no effect in this situation, because no plugin will be defined here. Am I right?